adr/util/memoize.py

```diff
@@ -17,14 +17,15 @@ def __call__(self, *args):
             self[args] = self.func(*args)
         return self[args]
 
-    def method_call(self, instance, *args):
+    def method_call(self, instance, *args, **kwargs):
         name = "_%s" % self.func.__name__
         if not hasattr(instance, name):
             setattr(instance, name, {})
         cache = getattr(instance, name)
-        if args not in cache:
-            cache[args] = self.func(instance, *args)
-        return cache[args]
+        kwargs_tuple = tuple(kwargs.items())
```

Comment:
Note: this will mean a call to the same function with the kwargs in a different order will not hit the cache, but I think that's OK (the same limitation applies to lru_cache in the standard lib)